core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -66,7 +73,51 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments, repository
   override def batchUpsertEntities(
     entityUpdates: Source[EntityUpdateDefinition, _],
     parentContext: RawlsRequestContext
-  ): Future[Source[Entity, _]] = ???
+  ): Future[Source[Entity, _]] = {
```

Comment:
I know that stream (Source) programming can be hard to read; I welcome any feedback on making this easier to follow and reverse-engineer.